ui/components/app/snaps/snap-icon/snap-icon.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FunctionComponent } from 'react';
+
+import { useSelector } from 'react-redux';
+import {
+  getSnapMetadata,
+  getTargetSubjectMetadata,
+} from '../../../../selectors';
+import { getAvatarFallbackLetter } from '../../../../helpers/utils/util';
+import {
+  AvatarBase,
+  AvatarBaseSize,
+  AvatarFavicon,
+  AvatarFaviconProps,
+  AvatarFaviconSize,
+  IconSize,
+} from '../../../component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  Display,
+  JustifyContent,
+  TextColor,
+} from '../../../../helpers/constants/design-system';
+
+type SnapIconProps = {
+  snapId: string;
+  avatarSize?: IconSize;
+  borderWidth?: number;
+  className?: string;
+  badgeBackgroundColor?: BackgroundColor;
+} & Omit<AvatarFaviconProps<'span'>, 'name'>;
+
+export const SnapIcon: FunctionComponent<SnapIconProps> = ({
+  snapId,
+  avatarSize = IconSize.Lg,
+  ...props
+}) => {
+  const subjectMetadata = useSelector((state) =>
+    getTargetSubjectMetadata(state, snapId),
+  );
+
+  const { name: snapName } = useSelector((state) =>
+    /* @ts-expect-error wrong type on selector. */
```

Comment:
Wrong type how? This comes from a JS file 🤔

Reply:
![image](https://github.com/user-attachments/assets/b770b4ec-fb4c-428c-bce9-d03937f54b65)